pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1722,30 +1723,97 @@ public void readEntryFailed(ManagedLedgerException exception, Object ctx) {
         });
     }
 
+    private CompletableFuture<Boolean> isNamespaceOperationAllowed(NamespaceName namespaceName,
```

Comment:
It seems like this method and the `isTopicOperationAllowed` method share a lot of core logic with just a couple places that would branch for looking up `allowNamespaceOperationAsync` vs `allowNamespaceOperationAsync`. It'd be good to consolidate this logic, if possible.